OpenRA.Mods.Common/Traits/BotModules/BaseBuilderBotModule.cs

```diff
@@ -193,10 +208,50 @@ void IBotPositionsUpdated.UpdatedDefenseCenter(CPos newLocation)
 
 		void IBotTick.BotTick(IBot bot)
 		{
+			// If we cannot solve pathfinding lag from AI's units stuck by AI's buildings,
+			// I actually not suggested to use this function, which increases the chance of
+			// just-built units being stuck on their way to rally point and cause game lag
 			SetRallyPointsForNewProductionBuildings(bot);
 
-			foreach (var b in builders)
-				b.Tick(bot);
+			BuildingsBeingProduced = new Dictionary<string, int>();
```

Comment:
I don't understand what this does. Is string an actor? What does the number mean?

Reply:
Yes. the string is the actor name and the int is the actor number. This var is for summarizing the buildings being produced in order to avoid producing multiple buildings.